scalr/data_source_scalr_ssh_key.go

```diff
@@ -0,0 +1,104 @@
+package scalr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/scalr/go-scalr"
+)
+
+func dataSourceScalrSSHKey() *schema.Resource {
+	return &schema.Resource{
+		Description: "Retrieves details of a specific SSH key by ID or name.",
+
+		ReadContext: dataSourceScalrSSHKeyRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description:  "ID of the SSH key.",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				AtLeastOneOf: []string{"name"},
+			},
+
+			"name": {
+				Description:  "Name of the SSH key.",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+
+			"account_id": {
+				Description: "ID of the account, in the format `acc-<RANDOM STRING>`.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				DefaultFunc: scalrAccountIDDefaultFunc,
+			},
+
+			"environments": {
+				Description: "List of environment IDs where the SSH key is available, or ``[\"*\"]` if shared.",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func dataSourceScalrSSHKeyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+
+	sshKeyID := d.Get("id").(string)
+	name := d.Get("name").(string)
+	accountID := d.Get("account_id").(string)
+
+	var sshKey *scalr.SSHKey
+	var err error
+
+	if sshKeyID != "" {
+		// Search by ID
+		sshKey, err = scalrClient.SSHKeys.Read(ctx, sshKeyID)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("error retrieving SSH key by ID: %v", err))
+		}
```

Comment:
Do we allow to specify both the `ID` and `Name` attributes? If yes, then we should additionally check here that the names do match.